File: 69-sqrtx/sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        for(int i=0;i<=x;i++){
            if(1LL*i*i==x || x<1LL*(i+1)*(i+1)){
                return i;
            }
        }
        return -1;
    }
};

File: 69-sqrtx/sqrtx.java
class Solution {
    public int mySqrt(int x) {
        if(x==0 || x==1) return x;
        long l = 1;
        long h = x;
        while(l<=h){
            long mid = l+(h-l)/2;
            long sq = mid*mid;
            if(sq==x){
                return (int)mid;
            }
            else if(sq>x){
                h = mid-1;
            }
            else l = mid+1;
        }
        return (int)h;
    }
}